Sources/Apollo/ApolloClient.swift

```diff
@@ -154,6 +153,124 @@ extension ApolloClient: ApolloClientProtocol {
                                       callbackQueue: queue,
                                       completionHandler: resultHandler)
   }
+
+#if swift(>=5.5.2)
+  @available(macOS 10.15, iOS 13, watchOS 8, tvOS 15, *)
+  public func fetch<Query>(query: Query, cachePolicy: CachePolicy, contextIdentifier: UUID?, queue: DispatchQueue) async throws -> GraphQLResult<Query.Data> where Query : GraphQLQuery {
+    return try await withCheckedThrowingContinuation { continuation in
```

Comment:
I think this implementation will crash if we set a cachePolicy of `.returnCacheDataAndFetch`: the client will first return the cached data (if present) and then proceed to return data from the network call, invalidating the invariant from `withCheckedThrowingContinuation`:

> [You must call a resume method exactly once on every execution path throughout the program.](https://developer.apple.com/documentation/swift/checkedcontinuation)

I didn't have time to verify this, sorry

Reply:
I think you're right about that unfortunately 😕. Since that's the case, async/await may not be a good fit for the library. I'll poke around a little more and update later.

Reply:
Yeah, we are planning on rethinking these APIs after the 1.0 release. It may be more suitable to using some sort of `Combine` stream as the return from these methods. The other alternative would be to break the `fetch` for `.returnCacheDataAndFetch` out into a separate method.

This is certainly going to be an issue for watchers, which will have to use `Combine` or some other method there.

None of those changes are backwards compatible though, which is why they are currently slated for 2.0.

Reply:
That's a really great catch @TizianoCoroneo! Thanks for making it known.